aviatrix/data_source_aviatrix_all_transit_gateways.go

```diff
@@ -516,349 +511,331 @@ func dataSourceAviatrixListAllTransitGateway() *schema.Resource {
 }
 
 func dataSourceAviatrixAllTransitGatewayRead(d *schema.ResourceData, meta interface{}) error {
-
-	if !d.Get("all_transit_gateway").(bool) {
-		return nil
-	}
-
 	client := meta.(*goaviatrix.Client)
 
-	TransitGatewayNameList, err := client.GetTransitGatewayNameList()
+	TransitGatewayList, err := client.GetTransitGatewayList()
 	if err != nil {
 		return err
 	}
 	var result []map[string]interface{}
-	for i := range TransitGatewayNameList {
-		transitGatewayName := TransitGatewayNameList[i]
-		gateway := &goaviatrix.Gateway{
-			GwName: transitGatewayName,
-		}
-		gw, err := client.GetGateway(gateway)
-
-		if err != nil {
-			if err == goaviatrix.ErrNotFound {
-				continue
-			}
-			return fmt.Errorf("couldn't find Aviatrix Transit Gateway: %s", err)
-		}
-		if gw != nil {
-			transitGateway := make(map[string]interface{})
-			transitGateway["cloud_type"] = gw.CloudType
-			transitGateway["account_name"] = gw.AccountName
-			transitGateway["gw_name"] = gw.GwName
-			transitGateway["subnet"] = gw.VpcNet
-			if goaviatrix.IsCloudType(gw.CloudType, goaviatrix.AWSRelatedCloudTypes) {
-				transitGateway["vpc_id"] = strings.Split(gw.VpcID, "~~")[0]
-				transitGateway["vpc_reg"] = gw.VpcRegion
-				if gw.AllocateNewEipRead {
-					transitGateway["allocate_new_eip"] = true
-				} else {
-					transitGateway["allocate_new_eip"] = false
-				}
-			} else if goaviatrix.IsCloudType(gw.CloudType, goaviatrix.GCPRelatedCloudTypes) {
-				transitGateway["vpc_id"] = gw.VpcID
-				transitGateway["vpc_reg"] = gw.GatewayZone
-				if gw.AllocateNewEipRead {
-					transitGateway["allocate_new_eip"] = true
-				} else {
-					transitGateway["allocate_new_eip"] = false
-				}
-			} else if goaviatrix.IsCloudType(gw.CloudType, goaviatrix.AzureArmRelatedCloudTypes|goaviatrix.OCIRelatedCloudTypes) {
-				transitGateway["vpc_id"] = gw.VpcID
-				transitGateway["vpc_reg"] = gw.VpcRegion
+	for i := range TransitGatewayList {
+		gw := TransitGatewayList[i]
+		transitGateway := make(map[string]interface{})
+		transitGateway["cloud_type"] = gw.CloudType
+		transitGateway["account_name"] = gw.AccountName
+		transitGateway["gw_name"] = gw.GwName
+		transitGateway["subnet"] = gw.VpcNet
+		if goaviatrix.IsCloudType(gw.CloudType, goaviatrix.AWSRelatedCloudTypes) {
+			transitGateway["vpc_id"] = strings.Split(gw.VpcID, "~~")[0]
+			transitGateway["vpc_reg"] = gw.VpcRegion
+			if gw.AllocateNewEipRead {
```

Comment:
This if statement is not needed. You can just set `transitGateway["allocate_new_eip"] = gw.AllocateNewEipRead`